src/wmtk/operations/TriMeshSplitEdgeAtMidpointOperation.cpp

```diff
@@ -0,0 +1,54 @@
+#include "TriMeshSplitEdgeAtMidpointOperation.hpp"
+
+#include <wmtk/TriMesh.hpp>
+#include "TriMeshSplitEdgeOperation.hpp"
+
+namespace wmtk {
+TriMeshSplitEdgeAtMidpointOperation::TriMeshSplitEdgeAtMidpointOperation(
+    Mesh& m,
+    const Tuple& t,
+    const OperationSettings<TriMeshSplitEdgeAtMidpointOperation>& settings)
+    : Operation(m)
+    , m_input_tuple{t}
+    , m_pos_accessor{m.create_accessor(settings.position)}
+    , m_min_squared_length{settings.min_squared_length}
+    , m_split_boundary_edges{settings.split_boundary_edges}
+{
+    p0 = m_pos_accessor.vector_attribute(m_input_tuple);
+    p1 = m_pos_accessor.vector_attribute(m_mesh.switch_vertex(m_input_tuple));
```

Comment:
I wrote it like that because the positions are used in `before` and `exectute`. I thought, I avoid accessing the attribute twice but I can also change it to local variables if that is preferred.